core/app/models/spree/order.rb

```diff
@@ -516,6 +516,9 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
+      # An unreturned exchange may have all of its shipments already shipped
+      return self.shipments if shipments.present? && shipments.all? { |s| s.shipped? }
```

Comment:
Is it possible to check that this is a "unreturned exchange" as well as doing this check?

I don't think that this "silent fail and skip" would make sense for a normal order.